### PR TITLE
feat: add scrollable video

### PR DESCRIPTION
**Pull request by YuCJ**
DEMO: Use `cd packages/scrollable-video; make dev-server` to start a dev server

### TODOs:

Cross devices supporting:
  - [x] Hide controls on iOS
  - [x] Fix fullscreen video on mobile devices
  - [x] Add responsive video (vertical and small video for mobile device)
  - [x] Fix blanking screen when scroll out the bottom of section on Safari
  - [x] Make sure this works on ~IE11~ and edge

### CODE REVIEW

**Comment by YuCJ** [comment]
The cross browser compatibility has been tested (except IE and Edge), it's ready for review now

File: packages/scrollable-video/docs/zh-tw/使用說明.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+# 「報導者／滾動式影片」產生器使用說明－－《報導者開放實驗室 The Reporter Lab》
+
+「報導者開放實驗室」的「新聞敘事元件庫」再釋出一個新的元件囉！這次推出的是「滾動式影片」，請看 DEMO：
+
+[DEMO]() <!-- TODO: Add demo -->
+
+不管是動畫、照片、插畫、圖表、錄影，只要把它剪輯成影片，就可以讓讀者一邊看影片，一邊閱讀提示文字。
+
+和一般影片的差別是，滾動式影片不會自動播放，而是在透過讀者滑動頁面的動作逐格步進，更接近使用觸控式載具的閱讀體驗。
```
Comment (nickhsine) [inline packages/scrollable-video/docs/zh-tw/使用說明.md:9]: `逐格播放`？

File: packages/scrollable-video/src/build-code/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+const _ = {
+  get,
+  map,
+}
+
+/**
+ *
+ * @export
+ * @param {Object} data - Data for Dual Channel Root React Component
```
Comment (nickhsine) [inline packages/scrollable-video/src/build-code/index.js:16]: `Data for Scrollable Video Root React Component`?

File: packages/scrollable-video/src/components/captions.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+ * @param {number} props.sectionHeight
+ * @param {number} props.viewportHeight
+ * @param {string} props.xBoxAlign - The global horizontal alignment of the caption box. Must be one of 'left', 'center', or 'right'.
+ * @param {string} props.xPosition - The global horizontal position of the captions relative to viewport.. Example:
+      { xBoxAlign: 'left', xPosition: '50px' } means the position is 50 px from the left edge of viewport to right
+      { xBoxAlign: 'center', xPosition: '50px' } means the position is 50 px from the center of viewport to right
+      { xBoxAlign: 'right', xPosition: '8.7%' } means the position is 8.7% (related to the section width) from the right edge of viewport to left
+ * @param {string} props.yBoxAlign - The global vertical alignment. Must be one of 'bottom' or 'top'.
+ * @param {string} props.yPosition - The global vertical position of the captions relative to viewport. Example:
+      { yBoxAlign: 'top', xPosition: '50px' } means the position is 50 px from the left edge of viewport to right
```
Comment (nickhsine) [inline packages/scrollable-video/src/components/captions.js:156]: `xPosition: '50px'` -> `yPosition: '50px'`?

Comment (nickhsine) [reply packages/scrollable-video/src/components/captions.js:156]: `left edge of viewport to right` -> `top edge of viewport to bottom` ?

File: packages/scrollable-video/src/components/captions.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+      { yBoxAlign: 'bottom', xPosition: '5%' } means the position is 5% (related to the section height, which is "viewport height * vhPerSecond * duration") from the bottom edge of viewport to top
```
Comment (nickhsine) [inline packages/scrollable-video/src/components/captions.js:157]: `xPosition: '50px'` -> `yPosition: '50px'`?

File: packages/scrollable-video/README.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+/**
+ * @typedef {Object} CaptionsSetting
+ * @property {string} xBoxAlign - The global horizontal alignment of the caption box. Must be one of 'left', 'center', or 'right'.
+ * @property {string} xPosition - The global horizontal position of the captions relative to viewport.. Example:
+      { xBoxAlign: 'left', xPosition: '50px' } means the position is 50 px from the left edge of viewport to right
+      { xBoxAlign: 'center', xPosition: '50px' } means the position is 50 px from the center of viewport to right
+      { xBoxAlign: 'right', xPosition: '8.7%' } means the position is 8.7% (related to the section width) from the right edge of viewport to left
+ * @property {string} yBoxAlign - The global vertical alignment. Must be one of 'bottom' or 'top'.
+ * @property {string} yPosition - The global vertical position of the captions relative to viewport. Example:
+      { yBoxAlign: 'top', xPosition: '50px' } means the position is 50 px from the left edge of viewport to right
```
Comment (nickhsine) [inline packages/scrollable-video/README.md:36]: `xPosition: '50px'` -> `yPostion: '50px'`?
`left edge of viewport to right` -> `top edge of viewport to bottom` ?

File: packages/scrollable-video/README.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+      { yBoxAlign: 'bottom', xPosition: '5%' } means the position is 5% (related to the section height, which is "viewport height * vhPerSecond * duration") from the bottom edge of viewport to top
```
Comment (nickhsine) [inline packages/scrollable-video/README.md:37]: `xPosition: '5%'` -> `yPosition: '5%'`?

File: packages/scrollable-video/src/build-code/index.js (hunk 2 above)
Comment (nickhsine) [reply packages/scrollable-video/src/build-code/index.js:16]: Copy `README.md` JSDoc typedef to here?
```
@param {Data} data - Data for Scrollable Video Root React Component
```

File: packages/scrollable-video/src/sheets/index.js [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+      majorDimension: sheetMetadata.captions.majorDimension,
+    })
+    const captions = tableToJSONRecords({
+      keys: captionsData[sheetMetadata.captions.keysIndex],
+      valuesOfRecords: captionsData.slice(
+        sheetMetadata.captions.recordsIndexStart
+      ),
+      parseValues: (key, value) => {
+        if (key === 'time') {
+          return parseInt(value, 10)
```
Comment (nickhsine) [inline packages/scrollable-video/src/sheets/index.js:135]: `parseInt` -> `parseFloat`?

File: packages/scrollable-video/src/sheets/index.js [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+    const appProps =
+      Array.isArray(appPropsData) && appPropsData.length > 0
+        ? tableToJSONRecords({
+            keys: appPropsData[sheetMetadata.appProps.keysIndex],
+            valuesOfRecords: appPropsData.slice(
+              sheetMetadata.appProps.recordsIndexStart
+            ),
+            parseValues: (key, value) => {
+              if (key === 'secondPerVh') {
+                return parseInt(value, 10)
```
Comment (nickhsine) [inline packages/scrollable-video/src/sheets/index.js:183]: `parseInt` -> `parseFloat` ?

**Comment by nickhsine** [comment]
@YuCJ 
This PR is kind of huge.
I think we could review and update according to reviews concurrently.

**Pull request merged**